compiler/rustc_typeck/src/astconv/generics.rs

```diff
@@ -108,7 +108,19 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                         Applicability::MaybeIncorrect,
                     );
                 }
-            }
+            },
+            (
+                GenericArg::Const(cnst),
+                GenericParamDefKind::Type { .. }
+            ) => {
+                let body = tcx.hir().body(cnst.value.body);
+                if let rustc_hir::ExprKind::Path(rustc_hir::QPath::Resolved(_, path)) = body.value.kind {
+                    if let Res::Def(DefKind::Fn { .. }, _) = path.res {
+                        err.help(&format!("`{}` is a function item, not a type", /* what goes here */));
```

Comment:
I do not know how to get the name of the argument back.

Reply:
https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/context/struct.TyCtxt.html#method.item_name for the `DefId` in the `Res`